src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.WaitForChanged.cs

```diff
@@ -125,84 +125,133 @@ public void NonZeroTimeout_NoActivity_TimesOut(WatcherChangeTypes changeType, bo
         }
 
         [Theory]
-        [OuterLoop("This test has a longer than average timeout and may fail intermittently")]
         [InlineData(WatcherChangeTypes.Created)]
         [InlineData(WatcherChangeTypes.Deleted)]
         public void CreatedDeleted_Success(WatcherChangeTypes changeType)
         {
             using (var testDirectory = new TempDirectory(GetTestFilePath()))
             using (var fsw = new FileSystemWatcher(testDirectory.Path))
             {
-                Task<WaitForChangedResult> t = Task.Run(() => fsw.WaitForChanged(changeType, LongWaitTimeout));
-                while (!t.IsCompleted)
+                for (int i = 1; i <= DefaultAttemptsForExpectedEvent; i++)
                 {
-                    string path = Path.Combine(testDirectory.Path, Path.GetRandomFileName());
-                    File.WriteAllText(path, "text");
-                    Task.Delay(BetweenOperationsDelayMilliseconds).Wait();
-                    if ((changeType & WatcherChangeTypes.Deleted) != 0)
+                    Task<WaitForChangedResult> t = Task.Run(() => fsw.WaitForChanged(changeType, WaitForExpectedEventTimeout));
+                    while (!t.IsCompleted)
                     {
-                        File.Delete(path);
+                        string path = Path.Combine(testDirectory.Path, Path.GetRandomFileName());
+                        File.WriteAllText(path, "text");
+                        Task.Delay(BetweenOperationsDelayMilliseconds).Wait();
+                        if ((changeType & WatcherChangeTypes.Deleted) != 0)
+                        {
+                            File.Delete(path);
+                        }
                     }
-                }
 
-                Assert.Equal(TaskStatus.RanToCompletion, t.Status);
-                Assert.Equal(changeType, t.Result.ChangeType);
-                Assert.NotNull(t.Result.Name);
-                Assert.Null(t.Result.OldName);
-                Assert.False(t.Result.TimedOut);
+                    if ((TaskStatus.RanToCompletion != t.Status) ||
+                        (changeType != t.Result.ChangeType) ||
+                        (null == t.Result.Name) ||
+                        (null != t.Result.OldName) ||
+                        (t.Result.TimedOut))
+                    {
+                        if (i == DefaultAttemptsForExpectedEvent)
+                        {
+                            Assert.Equal(TaskStatus.RanToCompletion, t.Status);
+                            Assert.Equal(changeType, t.Result.ChangeType);
+                            Assert.NotNull(t.Result.Name);
+                            Assert.Null(t.Result.OldName);
+                            Assert.False(t.Result.TimedOut);
+                        }
```

Comment:
Rather than duplicating the assert logic, how about something like:
```C#
try
{
   ... // existing asserts
}
catch
{
    if (i == DefaultAttemptsForExpectedEvents) throw;
}
```
?